### PR TITLE
automatically retry 10 times with pauses when the server sends a 429

### PR DESCRIPTION
**Pull request by chregu**
The rokka API sends now a 429 if the client sends too many requests (doc about that is soon ready). This implements retrying, when that happens

(not sure it's right way to do, feedback welcome ;))

### CODE REVIEW

**Comment by pstadler** [comment]
A couple of remarks:

- `fs` only works on the server, not in the browser. Is it necessary to re-read files?
- The code style does not match the rest of the repository
- This check-defined library seems to be deprecated / not widely used. The following helper which should be placed in `utils.js` does the same:

```
export function hasProperty(obj, key) {
  return key.split('.').every((k) => {
    if (typeof obj !== 'object' || obj === null || !k in obj) {
      return false;
    }

    obj = obj[k];

    return true;
  });
}
```

**Comment by pstadler** [comment]
It would be nice to make those values configurable.

File: src/transport.js [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+const retryRequest = (promiseRequest) => {
+  return req => {
+
+    return promiseRetry(
+      retry => {
+
+        return promiseRequest( req )
+          .catch(err => {
+            if (err.statusCode === 429) {
+              retry(err);
```
Comment (pstadler) [inline src/transport.js]: return here

File: src/transport.js [hunk 2]
```diff
@@ -1,3 +1,32 @@
 import request from 'request-promise';
+import promiseRetry from 'promise-retry';
 
-export default request;
+// When the API sends a 429 back, retry it for 10 times
+// This code is mainly copied from
+// https://github.com/CanTireInnovations/request-promise-retry/blob/master/index.js
+
+const retryRequest = (promiseRequest) => {
+  return req => {
```
Comment (pstadler) [inline src/transport.js]: `(req, retryOptions) => promiseRetry(...`

And later on:
`Object.assign(defaults, retryOptions);`

File: README.md [hunk 3]
```diff
@@ -39,6 +39,8 @@ var rokka = require('rokka')({
   apiVersion: <number>, // default: 1
   renderHost: '<url>',  // default: https://{organization}.rokka.io
   debug: true           // default: false
+  retryOptions: {}      // default: {retries: 10, minTimeout: 1000,
```
Comment (chregu) [inline README.md]: mmh, the problem with "transport" is that minTimeout and maxTimeout may not be clear, that it's for "retries" and not for general requests.

We can discuss it personally, what's the best approach when you have a minute

**Pull request merged**